andela/Ah-backend-guardians
# #162685678 Users should be able to report an article

#### What does this PR do?
This PR enables authorised users to report an article

#### Description of Task to be completed?
As an authenticated user, I should be able to report articles that violate terms of the agreement or are plagiarised to the authors haven admin for removal and or any other actions as per the terms of engagement

#### What are the relevant Pivotal Tracker stories?
#162685678[https://www.pivotaltracker.com/n/projects/2232496]

#### How should this be manually tested?
1. Using postman

[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/64596465cc2f6207c665)

or

2. Run each of the following commands in your terminal in the respective order.

- git clone https://github.com/andela/Ah-backend-guardians.git
- view the cloned repo in VS code
- In the terminal of VSCode, git checkout ft-report-an-article-162685678 
- Activate the virtual environment. (.venv/bin/activate)
- Run the app python manage.py runserver
- Lauch postman

- Hit end point http://127.0.0.1:8000/api/users/ to register a user
use parameters 
{
	"username":"pault",
	"email": "paulviv1@gmail.com",
	"password":"nabuloff123W"
}
- Hit end point http://127.0.0.1:8000/api/users/login to login a user
{
	"email": "paulviv1@gmail.com",
	"password":"nabuloff123W"
}
- Authenticate the user by passing the generated token in the header or place the token in the environment variables
Hit this URL to access the create report endpoint. post http://127.0.0.1:8000/api/articles/slug/report/
Use these parameters
{
	"reason": "Tribalistic"
}
You should get this result

![image](https://user-images.githubusercontent.com/41106626/52030098-426e7c00-2527-11e9-851b-60832a91ee35.png)
- Hit this URL to access the a list of reported articles by the current authorised user.
get http://127.0.0.1:8000/api/articles/slug/report/
- Hit this URL to access the getReportedArticles by a non-staff member endpoint. 
get http://127.0.0.1:8000/api/articles/slug/report_list/

You should get this result

![image](https://user-images.githubusercontent.com/41106626/52030155-8feae900-2527-11e9-8483-7ae02ec9ecc4.png)

- In the terminal create a superuser

You may use these parameters
{
         "username":"nabulo",
	"email": "nabulovivian2014@gmail.com",
	"password":"vivian@2018"
}
- Login using the super user crudentials
- Hit this URL to access the getReportedArticles by an admin. http://127.0.0.1:8000/api/articles/slug/report_list/

You should get this result

![image](https://user-images.githubusercontent.com/41106626/52030278-28816900-2528-11e9-8539-cd0d7d1dac13.png)
